macros/sql/union.sql

```diff
@@ -58,8 +58,7 @@
 
                     {%- set col = column_superset[col_name] %}
                     {%- set col_type = column_override.get(col.column, col.data_type) %}
-                    {%- set col_name = adapter.quote(col_name) if col_name in table_columns[table] else 'null' %}
-                    cast({{ col_name }} as {{ col_type }}) as {{ col.quoted }} {% if not loop.last %},{% endif %}
+                    {%- set col_name = adapter.quote(table.name) + '.' + adapter.quote(col_name) if col_name in table_columns[table] else 'null' %}                    cast({{ col_name }} as {{ col_type }}) as {{ col.quoted }} {% if not loop.last %},{% endif %}
```

Comment:
Can we rewrite this as:

```jinja2
{%- set col_name - %}
  {{ adapter.quote(table.name) ~ '.' ~ adapter.quote(col_name) }} 
{%- else -%}
  null
{% endif %}
cast({{ col_name }} as {{ col_type }}) as {{ col.quoted }} {% if not loop.last %},{% endif %}
```

I actually think we're better off just always including the table name in the `col_name` variable - that makes more sense to me than including it conditionally based on the supplied list of tables.

This is one of those things that I'd need to think a lot harder about though. Different databases have different semantics for things like this and I'm afraid of breaking eg. Snowflake in pursuit of fixing BigQuery.

Just for the sake of discussion: we might be able to alternatively alias the table name in the `from` clause of each `union` query. That might accomplish something similar.

I'll poke around with this some more - just wanted to get you some initial feedback here.